backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/format/impl/Elasticsearch6DefaultFieldFormatProvider.java

```diff
@@ -58,15 +58,22 @@
 		map.put( LocalDateTime.class, asImmutableList( "yyyy-MM-dd'T'HH:mm:ss.SSS", "yyyyyyyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS" ) );
 		map.put( OffsetTime.class, asImmutableList( "HH:mm:ss.SSSZZ", "HH:mm:ss.SSSSSSSSSZZ" ) );
 		map.put( OffsetDateTime.class, asImmutableList( "yyyy-MM-dd'T'HH:mm:ss.SSSZZ", "yyyyyyyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZZ" ) );
-		// ZoneRegionId is optional for ZonedDateTime, but we need the offset to handle ambiguous date/times at DST overlap
-		map.put( ZonedDateTime.class, asImmutableList( "yyyy-MM-dd'T'HH:mm:ss.SSSZZ'['ZZZ']'", "yyyyyyyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZZ'['ZZZ']'" ) );
-		map.put( Year.class, asImmutableList( "yyyy", "yyyyyyyyy" ) );
-		map.put( YearMonth.class, asImmutableList( "yyyy-MM", "yyyyyyyyy-MM" ) );
 		/*
-		 * This seems to be the ISO-8601 format for dates without year.
-		 * It's also the default format for Java's MonthDay, see MonthDay.PARSER.
+		 * ZoneRegionId is optional for ZonedDateTime, but we need the offset to handle ambiguous date/times at DST overlap.
+		 *
+		 * Also, in some cases the ZoneId might actually be a ZoneOffset (since ZoneOffset extends ZoneId),
+		 * so we need an alternative format to support this.
 		 */
-		map.put( MonthDay.class, asImmutableList( "--MM-dd" ) );
+		map.put( ZonedDateTime.class, asImmutableList(
+				"yyyy-MM-dd'T'HH:mm:ss.SSSZZ'['ZZZ']'",
+				"yyyyyyyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZZ'['ZZZ']'",
+				"yyyyyyyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZZ'['ZZ']'"
+		) );
+		map.put( Year.class, asImmutableList( "yyyy", "yyyyyyyyy" ) );
+		map.put( YearMonth.class, asImmutableList( "yyyy-MM", "yyyyyyyyy-MM" ) );
+		map.put( MonthDay.class, asImmutableList( "yyyy-MM-dd" ) );
```

Comment:
Do we have 2 put with the same key `MonthDay.class`?

Reply:
Good catch, I'll remove one.